src/frontend/apps/desk/src/features/teams/api/useCreateTeam.tsx

```diff
@@ -21,7 +18,12 @@ export const createTeam = async (name: string): Promise<CreateTeamResponse> => {
   });
 
   if (!response.ok) {
-    throw new Error(`Couldn't create team: ${response.statusText}`);
+    const error: APIError = {
+      status: response.status,
+      causes: await errorCauses(response),
+    };
```

Comment:
Won't it be better to encapsulate this duplicated code in your util function?
```suggestion
    throw await formatAPIError(response)
```

and `formatAPIError` returns an `APIError` instance?

Reply:
[86f5845](https://github.com/numerique-gouv/people/pull/52/commits/86f58452174e8c9394e6bd5c84ef0c7600ceed6b)